File: 0118-pascals-triangle/solution.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        if numRows == 1:
            return [[1]]
        elif numRows == 2:
            return [[1], [1, 1]]
        else:
            new = [1]
            result = self.generate(numRows-1)
            for i in range(len(result[-1])-1):
                new.append(result[-1][i] + result[-1][i+1])
            new.append(1)
            result.append(new)
            return result

File: 0118-pascals-triangle/solution.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) 
    {
        if(numRows == 0)
            return {{}};
        if(numRows == 1)
            return {{1}};
        else if(numRows == 2)
            return {{1}, {1, 1}};
        
        vector<vector<int>> ans = {{1},{1,1}};        
        for(int row = 2; row < numRows; row++)
        {
            int lastRow     = row - 1;
            int prevRowSize = ans[lastRow].size();
            vector<int> newRow(prevRowSize + 1, 1);
            
            for(int col = 1; col < prevRowSize; col++)
            {
                newRow[col] = ans[lastRow][col -1] + ans[lastRow][col];
            }
            
            ans.push_back(newRow);
        }
        
        return ans;
        
    }
};
